astropy/coordinates/builtin_frames/cirs_observed_transforms.py

```diff
@@ -90,9 +90,6 @@ def observed_to_cirs(observed_coo, cirs_frame):
     return cirs_at_aa_time.transform_to(cirs_frame)
 
 
-@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, AltAz, AltAz)
-@frame_transform_graph.transform(FunctionTransformWithFiniteDifference, HADec, HADec)
-def observed_to_observed(from_coo, to_frame):
-    # for now we just implement this through CIRS to make sure we get everything
-    # covered
-    return from_coo.transform_to(CIRS(obstime=from_coo.obstime)).transform_to(to_frame)
+# Create loopback transformations
+frame_transform_graph._add_merged_transform(AltAz, CIRS, AltAz)
```

Comment:
following #12203, can we route this via ICRS, and add a comment so it is not regressed?

Reply:
Done!